lib/events-reader.js

```diff
@@ -131,19 +163,11 @@ module.exports = function (harvesterApp) {
 
                                 var changeHandlerOp = opMap[op];
                                 var opFn = changeHandler[changeHandlerOp];
-                                if (opFn) {
-                                    debug('processing resource op ' + changeHandlerOp);
-                                    new RSVP.Promise(function (resolve, reject) {
-                                        resolve(opFn(id));
-                                    })
-                                        .then(function () {
-                                            dfd.resolve(doc);
```

Comment:
Alternatively, you could pass doc, o, o2. 

A similar filter would look like: 

        .onChange('whatever',{update:function(id, o, o2){
            var filter = '$set.equipment';
            var test = _.has(doc, filter, false);
......
        }});